qiskit/wrapper/__init__.py

```diff
@@ -0,0 +1,151 @@
+            group (str): The group used for online backend.
+            project (str): The project used for online backend.
+            proxies (dict): Proxy configuration for the API, as a dict with
+                'urls' and credential keys.
+            verify (bool): If False, ignores SSL certificates errors.
+            provider_name (str): the unique name for the online backend
+                provider (for example, 'qiskit' for the IBM Quantum Experience).
+    Raises:
+        QISKitError: if the provider name is not recognized.
+    """
+    if provider_name == 'qiskit':
+        provider = IBMQProvider(token, url,
+                                hub, group, project, proxies, verify)
+        _DEFAULT_PROVIDER.add_provider(provider)
+    else:
+        raise QISKitError('provider name %s is not recognized' % provider_name)
+
+# Functions for inspecting and retrieving backends.
+
+
+def available_backends(filters=None):
+    """
+    Return the backends that are available in the SDK, optionally filtering
+    them based on their capabilities.
+
+    Note:
+        In order for this function to return online backends, a connection with
+        an online backend provider needs to be established by calling the
+        `register()` function.
+    Args:
+        filters (dict): dictionary of filtering conditions.
+    Returns:
+        list[str]: the names of the available backends.
+    """
+    return _DEFAULT_PROVIDER.available_backends(filters)
+
+
+def local_backends():
+    """
+    Return the available local backends.
+
+    Returns:
+        list[str]: the names of the available remote backends.
+    """
+    return available_backends({'local': True})
+
+
+def remote_backends():
+    """
+    Return the available remote backends.
+
+    Returns:
+        list[str]: the names of the available remote backends.
+    """
+    return available_backends({'local': False})
+
+
+def get_backend(name):
+    """
+    Return an instance of a `Backend` object from its identifier.
+
+    Args:
+        name(str): unique name of the backend.
+    Returns:
+        BaseBackend: a Backend instance.
+    """
+    return _DEFAULT_PROVIDER.get_backend(name)
+
+
+# Functions for compiling and executing.
+
+
+def compile(list_of_circuits, backend_id, compile_config=None):
+    """Compile a list of circuits into a qobj.
+
+    Args:
+        list_of_circuits (list[QuantumCircuits]): list of circuits
+        backend_id (str): a backend name to use as the default compiling option
+        compile_config (dict or None): a dictionary of compile configurations.
+            If `None`, the default compile configuration will be used.
+    Returns:
+        obj: the qobj to be run on the backends
+    """
+    # pylint: disable=redefined-builtin
+    backend = _DEFAULT_PROVIDER.get_backend(backend_id)
+    return qiskit._compiler.compile(list_of_circuits, backend, compile_config)
+
+
+def execute(list_of_circuits, backend_id, compile_config=None,
```

Comment:
1. Yes, clarity was the goal - I feel it is better to be explicit, and reserve `backend` for actual instances. `backend_name` sounds good (I prefer `id` as it places more emphasis on it being a "unique identifier", but either is fine).

2. I'm on the same page - it is hard to declare and document (it touches the whole issue of using dicts). Since a small revision of `qiskit._compile` should be done after this PR, I'm hoping we can tackle the `compile_config` dict as a whole during that one and update the wrapper function accordingly if needed.

Reply:
ok i agree